### PR TITLE
Fix the makefile so GME is included during the compiling process on linux builds

### PR DESCRIPTION
**Pull request by SteelTitanium**
As posted here. http://mb.srb2.org/showthread.php?p=784168 by ForgiveThisNewb, GME isn't being included in the compiling process, which disables GME playback support and that's due to how the makefile was setup, this merge request fixes that problem and detects GME properly now. I have tested the changes and no problems found compiling the game.

### CODE REVIEW

File: src/Makefile [hunk 1]
```diff
@@ -315,12 +318,12 @@ LIBS+=$(PNG_LDFLAGS)
 CFLAGS+=$(PNG_CFLAGS)
 endif
 
-ifdef HAVE_LIBGME
+ifndef NOGME
 OPTS+=-DHAVE_LIBGME
 
 LIBGME_PKGCONFIG?=libgme
 LIBGME_CFLAGS?=$(shell $(PKG_CONFIG) $(LIBGME_PKGCONFIG) --cflags)
-LIBGME_LDFLAGS?=$(shell $(PKG_CONFIG) $(LIBGME_PKGCONFIG) --libs)
```
Comment (alama) [inline src/Makefile]: why the -lz, if you going to using zlib, add LIBGME_LDFLAGS after LIBPNG_LDFLAGS, LIBPNG_LDFLAGS have the flags flags to link to zlib

Comment (alama) [inline src/Makefile]: why the -lz, if you going to using zlib, add LIBGME_LDFLAGS after LIBPNG_LDFLAGS, LIBPNG_LDFLAGS have the flags flags to link to zlib

Comment (SteelTitanium) [inline src/Makefile]: Oh, I did not know that. Will change that soon.

Comment (alama) [inline src/Makefile]: never, the old libpng12 used to do, but libpng16 does not anymore

use  pkg-config on zlib to get the cflags and libs flags

**Comment by alama** [comment]
ok, new branch, zlib_pkgconfig
to use pkgconfig to find the zlib flags
also have GME on by defaut for Linux systems

**Comment by SteelTitanium** [comment]
Well, since Alam made a new branch for this. I'll just close this PR now.

**Pull request closed without merging**